### DISCUSSION TITLE
Clerk in mobile app (Flutter/iOS/Kotlin)?

### DISCUSSION THREAD

**Discussion by depsimon** [clerk/javascript#2869] [Support request - DO NOT OPEN HERE]
We're on the market for a new Auth Provider and Clerk is checking a lot of marks for us.

We're pretty confident the integration in the web apps will be easy but we're worried about the mobile applications as there is no SDK for Flutter/Kotlin/iOS.

What's the easiest way to work with Clerk in a mobile app environment?

For instance would it be possible for instance to use the Clerk Account Portal to manage 80% of the job and then refresh sessions with the Frontend API? Or is the way to go building all the UI around the Frontend API?

**Comment by timreibe** [comment]
Hey @depsimon did you find more information about integrating clerk into your mobile apps? If you decided already, which auth provider did you choose?

**Reply by depsimon** [reply]
Hi @timreibe I did not. We upvoted the mobile SDKs on the [roadmap](https://feedback.clerk.com/roadmap) but in the meantime we had to choose another provider that our backend team was more comfortable with (Duende).

**Reply by timreibe** [reply]
Thanks for your reply, I'm also currently looking for an auth provider for a mobile app and will take a look at Duende. Did you include [Kinde](https://kinde.com) in your decision back then or do you have any experience with it?

**Reply by depsimon** [reply]
Haven't check Kinde. We narrowed our choices to Clerk & WorkOS but both lack mobile SDKs and WorkOS only has 3 social providers.

In the end we chose to self-host our Auth server as it's the only option that let us customize it to our needs.

**Reply by lutfi-haslab** [reply]
@timreibe you can check out auth0, it has lot mobile integration. My first option was Clerk, since clerk mobile only support React Native Expo, then i am using another provider.

**Reply by TravisPageV1** [reply]
@lutfi-haslab Auth0 is great, but also quite literally 100x the price of Clerk.

**Reply by jescalan** [reply]
We have a flutter sdk coming out in beta shortly. If anyone would like early access feel free to reach out. We have a stable expo sdk as well that can build mobile apps.

**Comment by jescalan** [comment]
As noted in the discussion, we're working on mobile SDKs and you can check our roadmap to vote and see the status. We have support for mobile via expo, an iOS native SDK in beta, and we are beginning to work on Flutter support shortly!

**Reply by journeyrewardsjohn** [reply]
Would be great to see this pushed up for Kotlin too...having iOS is great, but most at scale apps are cross-platform

**Comment by fallenreaper** [comment]
Bump

Was looking around and started to use your stuff, but I wanted to see if it is possible to integrate with Kotlin Multiplatform so that my clients would verify on their devices etc.  Is there a plan to extend this soon?  Would this be something you would want someone to implement and submit a PR for?
